examples/pubsub/tutorial_pubsub_mqtt_subscribe.c

```diff
@@ -0,0 +1,329 @@
+ * the system preconfiguration and connection can be found in
+ * ``tutorial_pubsub_connection.c``.
+ */
+
+#include "open62541/server.h"
+#include "open62541/server_config_default.h"
+#include "ua_pubsub.h"
+#include "ua_network_pubsub_mqtt.h"
+#include "open62541/plugin/log_stdout.h"
+#include <signal.h>
+
+#define CONNECTION_NAME              "MQTT Subscriber Connection"
+#define TRANSPORT_PROFILE_URI        "http://opcfoundation.org/UA-Profile/Transport/pubsub-mqtt"
+#define MQTT_CLIENT_ID               "TESTCLIENTPUBSUBMQTT_SUB"
+#define CONNECTIONOPTION_NAME        "mqttClientId"
+#define TOPIC              			"customTopic"
+#define BROKER_ADDRESS_URL           "opc.mqtt://127.0.0.1:1883"
+
+
+//static UA_Boolean useJson = false;
+static UA_NodeId connectionIdentifier;
+static UA_NodeId readerGroupIdentifier;
+static UA_NodeId readerIdentifier;
+
+static UA_DataSetReaderConfig readerConfig;
+void subscriber_message_callback(UA_ByteString *encodedBuffer, UA_ByteString *topic);
+
+void subscriber_message_callback(UA_ByteString *encodedBuffer, UA_ByteString *topic) {
+	UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "message received: topic=%s | size=%d.",
+	        topic->data, (int)encodedBuffer->length);
+	UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "%s\n.", encodedBuffer->data);
+}
+
+static UA_StatusCode
+addPubSubConnection(UA_Server *server, UA_NetworkAddressUrlDataType *networkAddressUrl) {
+    /* Details about the connection configuration and handling are located
+     * in the pubsub connection tutorial */
+
+    UA_PubSubConnectionConfig connectionConfig;
+    memset(&connectionConfig, 0, sizeof(connectionConfig));
+    connectionConfig.name = UA_STRING(CONNECTION_NAME);
+    connectionConfig.transportProfileUri = UA_STRING(TRANSPORT_PROFILE_URI);
+    connectionConfig.enabled = UA_TRUE;
+
+    /* configure address of the mqtt broker (local on default port) */
+    UA_Variant_setScalar(&connectionConfig.address, networkAddressUrl,
+                         &UA_TYPES[UA_TYPES_NETWORKADDRESSURLDATATYPE]);
+    /* Changed to static publisherId from tutorial_pubsub_mqtt_publish.c example*/
+    connectionConfig.publisherId.numeric = 2234;
+
+    /* configure options, set mqtt client id */
+    UA_KeyValuePair connectionOptions[1];
+    connectionOptions[0].key = UA_QUALIFIEDNAME(0, CONNECTIONOPTION_NAME);
+    UA_String mqttClientId = UA_STRING(MQTT_CLIENT_ID);
+    UA_Variant_setScalar(&connectionOptions[0].value, &mqttClientId, &UA_TYPES[UA_TYPES_STRING]);
+    connectionConfig.connectionProperties = connectionOptions;
+    connectionConfig.connectionPropertiesSize = 1;
+    UA_StatusCode retval = UA_Server_addPubSubConnection(server, &connectionConfig, &connectionIdentifier);
+
+    if (UA_STATUSCODE_GOOD != retval)
+    {
+    	return retval;
+    }else  {
+    	UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "addPubSubConnection successful.");
+    }
+
+    return retval;
+}
```

Comment:
Check only for UA_STATUSCODE_GOOD and remove else case. It's not needed because the method ends returns the retval anyway. Adjust if statement to the open62541 coding conventions.